File: python/find_and_validate.py
from nexez_agent_sdk import NexezApiError, create_client


def main() -> None:
    nexez = create_client(buyer_agent="nexez-python-example")

    matches = nexez.search("remote launch strategy consultant under 3000", limit=5, location="Remote")
    if not matches.get("results") or not matches["results"][0].get("offer"):
        print("No actionable Nexez offer found.")
        return

    first = matches["results"][0]
    slug = first["page"]["slug"]
    offer_key = first["offer"]["key"]

    page = nexez.get_agent_page(slug)
    offer = next((item for item in page.get("offers", []) if item.get("key") == offer_key), None)
    if not offer:
        print(f"Page {slug} loaded, but offer {offer_key} was not found.")
        return

    if offer.get("negotiation_action"):
        validation = nexez.validate_negotiation(
            slug=slug,
            offer=offer_key,
            query="Buyer wants a remote launch strategy engagement under 3000.",
            budget="USD 2500",
            timeline="next two weeks",
        )
        action = "negotiation"
    else:
        validation = nexez.validate_checkout(
            slug=slug,
            offer=offer_key,
            query="Buyer wants to validate checkout before booking.",
        )
        action = "checkout"

    print(
        {
            "page": slug,
            "offer": offer_key,
            "action": action,
            "validation_ok": validation.get("ok"),
            "dry_run": validation.get("dryRun", True),
        }
    )


if __name__ == "__main__":
    try:
        main()
    except NexezApiError as error:
        print({"error": str(error), "status": error.status, "body": error.body})

File: typescript/find-and-validate.ts
import { createNexezClient } from '@nexez/agent-sdk'

const nexez = createNexezClient({ buyerAgent: 'nexez-typescript-example' })

const matches = await nexez.search('remote launch strategy consultant under 3000', {
  location: 'Remote',
  limit: 5,
})

const first = matches.results[0]
if (!first?.offer) {
  console.log('No actionable Nexez offer found.')
  process.exit(0)
}

const slug = first.page.slug
const offerKey = first.offer.key
const page = await nexez.getAgentPage(slug)
const offer = page.offers.find((item) => item.key === offerKey)

if (!offer) {
  console.log(`Page ${slug} loaded, but offer ${offerKey} was not found.`)
  process.exit(0)
}

const supportsNegotiation = Boolean(offer.negotiation_action)
const validation = supportsNegotiation
  ? await nexez.validateNegotiation({
      slug,
      offer: offerKey,
      query: 'Buyer wants a remote launch strategy engagement under 3000.',
      budget: 'USD 2500',
      timeline: 'next two weeks',
    })
  : await nexez.validateCheckout({
      slug,
      offer: offerKey,
      query: 'Buyer wants to validate checkout before booking.',
    })

console.log({
  page: slug,
  offer: offerKey,
  action: supportsNegotiation ? 'negotiation' : 'checkout',
  validationOk: validation.ok,
})
